auth/src/main/java/io/grpc/auth/GcpServiceAccountIdentityCredentials.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.auth;
+
+import com.google.auth.oauth2.ComputeEngineCredentials;
+import com.google.auth.oauth2.IdTokenCredentials;
+import io.grpc.CallCredentials;
+import io.grpc.Status;
+import java.util.concurrent.Executor;
+
+/**
+ * {@link CallCredentials} that authenticates using the default service account
+ * of a Google Compute Engine (GCE) instance.
+ *
+ * <p>It obtains an ID token from the GCE metadata server and uses it to create
+ * {@link IdTokenCredentials} which are then adapted to {@link CallCredentials}
+ * using {@link MoreCallCredentials}.
+ *
+ * <p>This class is intended for use on GCE instances. It will not work
+ * in other environments.
+ */
+public class GcpServiceAccountIdentityCredentials extends CallCredentials {
```

Comment:
Don't make any new public APIs. You want to add code to xds and just use it in your filter.

GcpServiceAccountIdentityCallCredentials from the gRFC already exists in Java. That's what your `getCallCredentials()` function creates. You _absolutely_ wouldn't want to create that every RPC.

We will use custom `CallCredentials` in Java in order to read the `authority` and then select the correct `CallCredentials`, but that's going to look different from this.

Reply:
> GcpServiceAccountIdentityCallCredentials from the gRFC already exists in Java.

I cannot find it. Where is it?

Reply:
>> GcpServiceAccountIdentityCallCredentials from the gRFC already exists in Java.
>
> I cannot find it. Where is it?

The next sentence answered that:
> That's what your getCallCredentials() function creates.

C doesn't have an auth library to use, unlike Java and Go. So in C they will need to make a new class. In Java, the functionality is already available.